File: app/src/main/java/com/antyzero/smoksmog/settings/SettingsModule.java
package com.antyzero.smoksmog.settings;

import android.content.Context;

import com.antyzero.smoksmog.permission.PermissionHelper;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Singleton
@Module
public class SettingsModule {

    @Provides
    @Singleton
    @Deprecated
    public SettingsHelper provideSettingsHelper( Context context, PermissionHelper permissionHelper ) {
        return new SettingsHelper( context, permissionHelper );
    }
}

File: app/src/main/kotlin/com/antyzero/smoksmog/settings/SettingsModule.kt
package com.antyzero.smoksmog.settings

import android.content.Context
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Singleton
@Module
class SettingsModule {

    @Provides
    @Singleton
    fun provideSettingsHelper(context: Context) = SettingsHelper(context)
}
